javaslang/generator/Generator.scala

```diff
@@ -51,6 +51,302 @@ def generateMainClasses(): Unit = {
       val PredicateType = im.getType("java.util.function.Predicate")
       val SupplierType = im.getType("java.util.function.Supplier")
       val IteratorType = im.getType("javaslang.collection.Iterator")
+      val EitherType = im.getType("javaslang.control.Either")
+      val TryType = im.getType("javaslang.control.Try")
+      val ValidationType = im.getType("javaslang.control.Validation")
+      val FutureType = im.getType("javaslang.concurrent.Future")
+      val CheckedSupplierType = im.getType("javaslang.control.Try.CheckedSupplier")
+      val ExecutorServiceType = im.getType("java.util.concurrent.ExecutorService")
+      val ArrayType = im.getType("javaslang.collection.Array")
+      val CharSeqType = im.getType("javaslang.collection.CharSeq")
+      val VectorType = im.getType("javaslang.collection.Vector")
+      val ListType = im.getType("javaslang.collection.List")
+      val StreamType = im.getType("javaslang.collection.Stream")
+      val QueueType = im.getType("javaslang.collection.Queue")
+      val LinkedHashSetType = im.getType("javaslang.collection.LinkedHashSet")
+      val HashSetType = im.getType("javaslang.collection.HashSet")
+
+      def genTraversableAliases(traversableType: String, name: String): String = {
+        xs"""
+          /$javadoc
+           * Alias for {@code $traversableType.of()}
+           *
+           * @see $traversableType#of(T)
+           */
+          public static <T> $traversableType<T> $name(T element) {
+            return $traversableType.of(element);
+          }
+
+          /$javadoc
+           * Alias for {@code $traversableType.of()}
+           *
+           * @see $traversableType#of(T...)
+           */
+          @SuppressWarnings("varargs")
+          @SafeVarargs
+          public static <T> $traversableType<T> $name(T... elements) {
+            return $traversableType.of(elements);
+          }
+          """
+      }
+
+      def genAliases(im: ImportManager, packageName: String, className: String): String = {
+        xs"""
+          //
+          // Aliases for static factories
+          //
+
+          ${(0 to N).gen(i => {
+          val generics = (1 to i).gen(j => s"T$j")(", ")
+          val fullGenerics = s"<${(i > 0).gen(s"$generics, ")}R>"
+          xs"""
+              /$javadoc
+               * Alias for {@code Function$i.of()}
+               *
+               * @see Function$i#of(Function$i)
+               */
+              public static $fullGenerics Function$i$fullGenerics f$i(Function$i$fullGenerics methodReference) {
+                  return Function$i.of(methodReference);
+              }
+            """
+        })("\n\n")}
```

Comment:
minor/indentation: line 100-112: add two spaces, but line 111 needs to stay as-is :-)